sync/upstream.py

```diff
@@ -370,19 +370,19 @@ def try_land_pr(self):
         logger.info("Commit are landable; trying to land %s" % self.pr)
 
         msg = None
-        all_status_success, non_success = env.gh_wpt.status_checks_pass(
-            self.pr, exclude="upstream/gecko")
-        if not all_status_success:
-            if not any(item.state == "pending" for item in non_success.itervalues()):
-                details = ["Github PR %s" % env.gh_wpt.pr_url(self.pr)]
-                for context, item in non_success.iteritems():
-                    if item.state == "pending":
-                        continue
-                    url_str = " (%s)" % item.target_url if item.target_url else ""
-                    details.append("* %s%s" % (item.context, url_str))
-                details = "\n".join(details)
-                msg = ("Can't merge web-platform-tests PR due to failing upstream checks:\n%s" %
-                       details)
+        check_state, statuses = env.gh_wpt.get_combined_status(
+            self.pr, exclude={"upstream/gecko"})
+        non_success = [i for i in statuses if i.state != "success"]
+        if check_state not in ["success", "pending"]:
+            details = ["Github PR %s" % env.gh_wpt.pr_url(self.pr)]
+            for item in non_success:
+                if item.state == "pending":
```

Comment:
Is this actually possible now? Should it be "failing" instead?

Reply:
If the overall state is "failure" (check_state), then it's still possible to have "pending" in the list of individual statuses. (>=1 failure/error implies overall failure).

So what I'm going for is to report the failures/errors in the message details.